/// Files: drivers/net/ethernet/mellanox/mlx5/core/fpga/conn.c
/// Fix: c8c2a057fdc7de1cd16f4baa51425b932a42eb39
/// Fixes: 537a50574175a2b68b0612ffb48cb044a394c7b4

virtual detect

@err@
identifier in, err, err_cqwq;
position p;
@@

mlx5_fpga_conn_create_cq(...)
{
	...
*	in = kvzalloc(...);
	...
*	err = mlx5_vector2eqn(...);
*	if (err)
*		goto err_cqwq;@p
	... when any
}

@script:python depends on detect@
p << err.p;
@@

coccilib.report.print_report(p[0], 'ERROR: CVE-2019-19045')
